src/main/java/net/snowflake/ingest/utils/ParameterProvider.java

```diff
@@ -188,6 +200,13 @@ private void setParameterMap(Map<String, Object> parameterOverrides, Properties
         BDEC_PARQUET_COMPRESSION_ALGORITHM_DEFAULT,
         parameterOverrides,
         props);
+
+    // Required parameter override for Iceberg mode
+    if (this.isIcebergMode) {
+      this.parameterMap.put(
+          MAX_CHUNKS_IN_BLOB_AND_REGISTRATION_REQUEST,
```

Comment:
why not call this.updateValue here ? That's the only method that's updating parameterMap before this block was added, lets keep it that way..?

Reply:
The `updateValue` considers props and paramOverrides before default value. While under iceberg mode we should not allow user to set the chunk number > 1, that's why I use the `put` method directly. I think we can use `updateValue(key, value, null, null)` to achieve the same result, wdyt?